transport/kp/key_provider.go

```diff
@@ -210,6 +212,19 @@ func (sp *StandardProvider) Generate(algo string, size int) (err error) {
 		}
 		sp.internal.keyPEM = pem.EncodeToMemory(p)
 		sp.internal.priv = priv
+	case "ed25519":
```

Comment:
Here the size might be passed as argument to the function as well.. so for consistency, what do you think of checking it?